rfq/order.go

```diff
@@ -0,0 +1,320 @@
+		"event.")
+	acceptHtlcEvent := NewAcceptHtlcEvent(htlc, *channelRemit)
+	h.cfg.AcceptHtlcEvents <- acceptHtlcEvent
+
+	return &lndclient.InterceptedHtlcResponse{
+		Action: lndclient.InterceptorActionResume,
+	}, nil
+}
+
+// setupHtlcIntercept sets up HTLC interception.
+func (h *OrderHandler) setupHtlcIntercept(ctx context.Context) error {
+	// Intercept incoming HTLCs. This call passes the handleIncomingHtlc
+	// function to the interceptor. The interceptor will call this function
+	// in a separate goroutine.
+	err := h.cfg.HtlcInterceptor.InterceptHtlcs(ctx, h.handleIncomingHtlc)
+	if err != nil {
+		if fn.IsCanceled(err) {
+			return nil
+		}
+
+		return fmt.Errorf("unable to setup incoming HTLC "+
+			"interceptor: %w", err)
+	}
+
+	return nil
+}
+
+// mainEventLoop executes the main event handling loop.
+func (h *OrderHandler) mainEventLoop() {
+	log.Debug("Starting main event loop for order handler")
+
+	cleanupTicker := time.NewTicker(h.cfg.CleanupInterval)
+	defer cleanupTicker.Stop()
+
+	for {
+		select {
+		// Periodically clean up expired channel remits from our local
+		// cache.
+		case <-cleanupTicker.C:
+			log.Debug("Cleaning up any stale channel remits from " +
+				"the order handler")
+			h.cleanupStaleChannelRemits()
+
+		case <-h.Quit:
+			log.Debug("Received quit signal. Stopping negotiator " +
+				"event loop")
+			return
+		}
+	}
+}
+
+// Start starts the service.
+func (h *OrderHandler) Start() error {
+	var startErr error
+	h.startOnce.Do(func() {
+		log.Info("Starting subsystem: order handler")
+
+		// Start the main event loop in a separate goroutine.
+		h.Wg.Add(1)
+		go func() {
+			defer h.Wg.Done()
+
+			ctx, cancel := h.WithCtxQuitNoTimeout()
+			defer cancel()
+
+			startErr = h.setupHtlcIntercept(ctx)
+			if startErr != nil {
+				log.Errorf("Error setting up HTLC "+
+					"interception: %v", startErr)
+				return
+			}
+
+			h.mainEventLoop()
+		}()
+	})
+
+	return startErr
+}
+
+// RegisterChannelRemit registers a channel management remit. If a remit exists
+// for the channel SCID, it is overwritten.
+func (h *OrderHandler) RegisterChannelRemit(quoteAccept rfqmsg.Accept) error {
+	log.Debugf("Registering channel remit for SCID: %d",
+		quoteAccept.ShortChannelId())
+
+	channelRemit := NewChannelRemit(quoteAccept)
+
+	h.channelRemitsMtx.Lock()
+	defer h.channelRemitsMtx.Unlock()
+
+	h.channelRemits[channelRemit.Scid] = channelRemit
+
+	return nil
+}
+
+// FetchChannelRemit fetches a channel remit given a serialised SCID. If a
+// channel remit is not found, false is returned. Expired channel remits are
+// not returned and are removed from the cache.
+func (h *OrderHandler) FetchChannelRemit(scid SerialisedScid) (*ChannelRemit,
+	bool) {
+
+	h.channelRemitsMtx.Lock()
+	defer h.channelRemitsMtx.Unlock()
+
+	remit, ok := h.channelRemits[scid]
+	if !ok {
+		return nil, false
+	}
+
+	// If the remit has expired, return false and clear it from the cache.
+	if time.Now().Unix() > int64(remit.Expiry) {
```

Comment:
nit: maybe this would be slightly easier to read? `time.Unix(remit.Expiry, 0).Before(time.Now())`

Reply:
Latest commits contain:
```
	expireTime := time.Unix(int64(remit.Expiry), 0).UTC()
	currentTime := time.Now().UTC()

	if currentTime.After(expireTime) {
```

Let me know if that doesn't work for you.

Reply:
I don't think we need the explicit `.UTC()` call here, so this can be shortened into one line perhaps (or maybe not because of the `int64` cast). I think we only need to be careful and use `.UTC()` when we use the string representation of a date (which happens when storing a date in the DB for example).

Reply:
But non-blocking for this PR, let's get it merged!